### PR TITLE
Behat tests

### PR DESCRIPTION
**Pull request by GSadee**

### CODE REVIEW

File: features/addressing/managing_zones/deleting_multiple_zones.feature [hunk 1]
```diff
@@ -0,0 +1,19 @@
+@managing_zones
+Feature: Deleting multiple zones
+    In order to remove test, obsolete or incorrect zones
```
Comment (stefandoorn) [inline features/addressing/managing_zones/deleting_multiple_zones.feature]: Shouldn't it be something like: `In order to remove multiple zones in an efficient way`. Reason can vary for anyone.

And then also: `I want to be able to delete multiple zones at once`

File: features/addressing/managing_zones/deleting_multiple_zones.feature [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+    Background:
+        Given the store has zones "North America", "South America" and "Europe"
+        And I am logged in as an administrator
+
+    @ui @javascript
+    Scenario: Deleting multiple zones
+        When I browse zones
+        And I check the "North America" zone
+        And I check also the "South America" zone
+        And I delete them
```
Comment (stefandoorn) [inline features/addressing/managing_zones/deleting_multiple_zones.feature]: Maybe: `And I press the bulk delete button`?

File: src/Sylius/Behat/Context/Setup/CustomerGroupContext.php [hunk 3]
```diff
@@ -19,6 +19,7 @@
 use Sylius\Component\Customer\Model\CustomerGroupInterface;
 use Sylius\Component\Resource\Factory\FactoryInterface;
 use Sylius\Component\Resource\Repository\RepositoryInterface;
+use Webmozart\Assert\Assert;
```
Comment (stefandoorn) [inline src/Sylius/Behat/Context/Setup/CustomerGroupContext.php]: Not sure this one is needed in here.

File: src/Sylius/Behat/Context/Setup/PaymentContext.php [hunk 4]
```diff
@@ -101,15 +101,15 @@ public function storeAllowsPayingForAllChannels($paymentMethodName, array $chann
...
+     * @Given the store has (also) a payment method :paymentMethodName with a code :paymentMethodCode
      */
     public function theStoreHasAPaymentMethodWithACode($paymentMethodName, $paymentMethodCode)
     {
         $this->createPaymentMethod($paymentMethodName, $paymentMethodCode, 'Offline');
     }
 
     /**
-     * @Given the store has a payment method :paymentMethodName with a code :paymentMethodCode and Paypal Express Checkout gateway
+     * @Given the store has (also) a payment method :paymentMethodName with a code :paymentMethodCode and Paypal Express Checkout gateway
```
Comment (stefandoorn) [inline src/Sylius/Behat/Context/Setup/PaymentContext.php:112]: Do we need these?

File: src/Sylius/Behat/Context/Setup/PromotionContext.php [hunk 5]
```diff
@@ -96,10 +96,10 @@ public function __construct(
     }
 
     /**
-     * @Given there is a promotion :promotionName
+     * @Given there is (also) a promotion :promotionName
      * @Given there is a promotion :promotionName identified by :promotionCode code
      */
-    public function thereIsPromotion($promotionName, $promotionCode = null)
+    public function thereIsPromotion(string $promotionName, string $promotionCode = null): void
```
Comment (stefandoorn) [inline src/Sylius/Behat/Context/Setup/PromotionContext.php]: `?string $promotionCode = null`? Or does this work?

File: src/Sylius/Behat/Context/Setup/ZoneContext.php [hunk 6]
```diff
@@ -191,17 +203,17 @@ private function createZoneMember(CodeAwareInterface $zoneMember)
...
      *
      * @return ZoneInterface
      */
-    private function setUpZone($zoneName, $code, $scope)
+    private function createZone(string $name, ?string $code = null, ?string $scope = Scope::ALL): ZoneInterface
     {
         $zone = $this->zoneFactory->createTyped(ZoneInterface::TYPE_ZONE);
-        $zone->setCode($code);
-        $zone->setName($zoneName);
+        $zone->setCode($code ? $code : StringInflector::nameToCode($name));
```
Comment (stefandoorn) [inline src/Sylius/Behat/Context/Setup/ZoneContext.php]: `$code ?? StringInflector::nameToCode($name)`?

File: src/Sylius/Bundle/ResourceBundle/Controller/FlashHelper.php [hunk 7]
```diff
@@ -13,7 +13,9 @@
 
 namespace Sylius\Bundle\ResourceBundle\Controller;
 
+use Doctrine\Common\Inflector\Inflector;
 use Sylius\Bundle\ResourceBundle\Event\ResourceControllerEvent;
+use Sylius\Component\Resource\Metadata\MetadataInterface;
```
Comment (stefandoorn) [inline src/Sylius/Bundle/ResourceBundle/Controller/FlashHelper.php]: Do we need this one?

**Review by stefandoorn — CHANGES_REQUESTED** [review]

File: features/addressing/managing_zones/deleting_multiple_zones.feature (hunk 2 above)
Comment (GSadee) [reply features/addressing/managing_zones/deleting_multiple_zones.feature]: This is too precise for behat scenario (and not consistent with existent steps) in my opinion

File: src/Sylius/Behat/Context/Setup/PaymentContext.php (hunk 4 above)
Comment (GSadee) [reply src/Sylius/Behat/Context/Setup/PaymentContext.php:112]: I'm using that in `deleting_multiple_payment_methods.feature`

**Review by stefandoorn — APPROVED** [review]

**Pull request merged**